src/jobsautoreport/query.py

```diff
@@ -35,21 +36,73 @@ def _get_query_all_jobs(from_date: datetime, to_date: datetime) -> dict:
             }
         }
 
-    def query_all_jobs(
-        self, from_date: datetime, to_date: datetime
-    ) -> list[JobDetails]:
+    @staticmethod
+    def _get_query_all_steps(from_date: datetime, to_date: datetime) -> dict:
+        return {
+            "query": {
+                "bool": {
+                    "must": [
+                        {
+                            "match": {
+                                "step.name": {
+                                    "query": "baremetalds-packet-setup",
+                                    "operator": "and",
+                                }
+                            }
+                        },
+                        {
+                            "range": {
+                                "job.start_time": {
+                                    "gte": from_date,
+                                    "lte": to_date,
+                                }
+                            }
+                        },
+                    ]
+                }
+            }
+        }
+
+    def query_jobs(self, from_date: datetime, to_date: datetime) -> list[JobDetails]:
         query = self._get_query_all_jobs(from_date, to_date)
-        return self._query_and_log(query)
+        return self._query_jobs_and_log(query)
+
+    def query_step_events(
+        self, from_date: datetime, to_date: datetime
+    ) -> list[StepEvent]:
+        query = self._get_query_all_steps(from_date, to_date)
+        return self._query_step_events_and_log(query)
 
-    def _query_and_log(self, query: dict) -> list[JobDetails]:
+    def _query_jobs_and_log(self, query: dict) -> list[JobDetails]:
         logger.debug("OpenSearch query: %s", query)
-        jobs = self._scan(query)
-        return utils.parse_jobs(jobs)
+        objects = self._scan(query, self._jobs_index)
+        return self.parse_jobs(objects)
 
-    def _scan(self, query: dict[str, Any]) -> list[dict[Any, Any]]:
+    def _query_step_events_and_log(self, query: dict) -> list[StepEvent]:
+        logger.debug("OpenSearch query: %s", query)
+        objects = self._scan(query, self._steps_index)
```

Comment:
```suggestion
        es_steps = self._scan(query, self._steps_index)
```
try to not use much 'data', 'object' and all other too generic names